File: 2324-CS319/Week01/TimeAlg1.py
# Sum the elements of a list in Python
import time

N=10**7     # N=10^n
A = [1]*N
start = time.time()
s1=0;
for i in range(len(A)):
    s1+=A[i]
t1 = time.time() - start
print(f"N={N:6.0e}, error={s1-N}, time(s)={t1:6.2f}")
 

File: old/2324-CS319/Week01/TimeAlg1.cpp
// Sum the elements of a list in C++
#include <iostream>
#include <time.h>
#include <math.h>
int main() {
  int N=pow(10,6); // N=10^n
  double *A = new double [N];
  for (int i=0; i<N; i++)
    A[i]=1.0;
  clock_t start=clock();
  double s1=0;
  for (int i=0; i<N; i++)
    s1+=A[i];
  double num_clocks = (double)(clock()-start);
  double t1 = num_clocks/CLOCKS_PER_SEC;
  std::cout << "N=10^" << log10(N)
	    << ", error=" << s1-N
	    << ", time(s)" << t1 << std::endl;
  return(0);
}
 
